### PR TITLE
Move to standalone c_bindings: cpp_bindgen

### PR DESCRIPTION
**Pull request by havogt**
Removes the c_bindings from the repository and re-includes cpp_bindgen via `FetchContent()` in a backward-compatible name (i.e. wrappers for the old names). These wrappers are deprectated and will be removed in GridTools 2.0.

Users should not notice this change except for a small detail: previously users exported the GridTools c_bindings CMake targets (e.g. handle) with the explicit name, now the new CMake function `install_cpp_bindgen_targets()` should be used to avoid this problem in the future.

New users should avoid using the cpp_bindgen provided with the GridTools CMake, but should explicitly use the standalone version from [GridTools/cpp_bindgen](https://github.com/GridTools/cpp_bindgen).

### CODE REVIEW

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by fthaler** [comment]
launch perftest

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch icgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch icgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch icgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

File: cmake/GridToolsConfig.cmake.in [hunk 1]
```diff
@@ -37,7 +37,12 @@ if (GRIDTOOLS_ENABLE_PERFORMANCE_METERS)
...
 
 set_and_check(GridTools_MODULE_PATH @PACKAGE_GRIDTOOLS_MODULE_PATH@)
-set_and_check(GridTools_SOURCES_PATH @PACKAGE_GRIDTOOLS_SOURCES_PATH@)
 set_and_check(GridTools_INCLUDE_PATH @PACKAGE_GRIDTOOLS_INCLUDE_PATH@)
 
+# Remove with GT 2.0 (cpp_bindgen will be fully standalone)
+set(CPP_BINDGEN_GT_LEGACY ON CACHE BOOL "Enables the legacy mode for API compatibility with GridTools 1.x" FORCE)
+mark_as_advanced(CPP_BINDGEN_GT_LEGACY)
+find_dependency(cpp_bindgen
+    HINTS @GT_CPP_BINDGEN_CONFIG_LOCATION@) # TODO discuss with Lukas
```
Comment (havogt) [inline cmake/GridToolsConfig.cmake.in]: Here is a TODO

File: cmake/export.cmake [hunk 2]
```diff
@@ -5,28 +5,29 @@ include(CMakePackageConfigHelpers)
...
-  PATH_VARS GRIDTOOLS_MODULE_PATH GRIDTOOLS_SOURCES_PATH GRIDTOOLS_INCLUDE_PATH
+  PATH_VARS GRIDTOOLS_MODULE_PATH GRIDTOOLS_INCLUDE_PATH GT_CPP_BINDGEN_CONFIG_LOCATION
   INSTALL_DESTINATION ${PROJECT_BINARY_DIR})
 write_basic_package_version_file(
   ${PROJECT_BINARY_DIR}/GridToolsConfigVersion.cmake
-  COMPATIBILITY SameMinorVersion )
+  COMPATIBILITY SameMajorVersion )
 
-install(TARGETS gridtools EXPORT GridToolsTargets
+install(TARGETS gridtools cpp_bindgen_interface EXPORT GridToolsTargets
```
Comment (havogt) [inline cmake/export.cmake]:
```suggestion
install(TARGETS gridtools cpp_bindgen_interface EXPORT GridToolsTargets #TODO remove cpp_bindgen_interface in GT 2.0
```

File: cmake/export.cmake [hunk 3]
```diff
@@ -39,12 +40,12 @@ if (COMPONENT_GCL)
       RUNTIME DESTINATION bin
       INCLUDES DESTINATION include
     )
-    export(TARGETS gridtools gcl
+    export(TARGETS gridtools gcl cpp_bindgen_interface
```
Comment (havogt) [inline cmake/export.cmake]:
```suggestion
    export(TARGETS gridtools gcl cpp_bindgen_interface #TODO remove cpp_bindgen_interface in GT 2.0
```

File: cmake/export.cmake [hunk 4]
```diff
@@ -39,12 +40,12 @@ if (COMPONENT_GCL)
...
         FILE ${PROJECT_BINARY_DIR}/GridToolsTargets.cmake
         NAMESPACE GridTools::
     )
 else()
-    export(TARGETS gridtools
+    export(TARGETS gridtools cpp_bindgen_interface
```
Comment (havogt) [inline cmake/export.cmake]:
```suggestion
    export(TARGETS gridtools cpp_bindgen_interface #TODO remove cpp_bindgen_interface in GT 2.0
```

**Comment by havogt** [comment]
launch strgrid

File: CMakeLists.txt [hunk 5]
```diff
@@ -131,4 +125,5 @@ if(GT_ENABLE_PYUTILS)
     add_subdirectory(pyutils)
 endif()
 
-include(export)
+# absolute path to avoid collision with "export" from other packages
+include(${CMAKE_CURRENT_LIST_DIR}/cmake/export.cmake)
```
Comment (lukasm91) [inline CMakeLists.txt]: We should think about how we want to do this... but not in this PR.

File: include/gridtools/interface/repository/repository.hpp [hunk 6]
```diff
@@ -27,6 +26,7 @@
 #include <boost/preprocessor/selection/max.hpp>
 #include <boost/preprocessor/selection/min.hpp>
 #include <boost/preprocessor/tuple.hpp>
+#include <cpp_bindgen/export.hpp>
 
 #ifndef GT_PARSE_PREPROCESSOR
```
Comment (lukasm91) [inline include/gridtools/interface/repository/repository.hpp]: this is never defined -> remove

File: include/gridtools/interface/repository/repository.hpp [hunk 7]
```diff
@@ -14,7 +14,6 @@
     "GRIDTOOLS ERROR=> For the repository you need to \"#define BOOST_PP_VARIADICS 1\" before the first include of any boost preprocessor file.")
 #endif
 
-#include "../../c_bindings/export.hpp"
 #include "../../common/boost_pp_generic_macros.hpp"
 #include "../fortran_array_adapter.hpp"
 #include "boost/preprocessor/seq.hpp"
```
Comment (lukasm91) [inline include/gridtools/interface/repository/repository.hpp]: can you fix the includes here?
```suggestion
#include <boost/preprocessor/seq.hpp>
```

File: cmake/GridToolsConfig.cmake.in (hunk 1 above)
Comment (lukasm91) [reply cmake/GridToolsConfig.cmake.in]: ok for me

Comment (havogt) [inline cmake/GridToolsConfig.cmake.in]:
```suggestion
    HINTS @GT_CPP_BINDGEN_CONFIG_LOCATION@)
```

File: include/gridtools/interface/repository/repository.hpp (hunk 6 above)
Comment (havogt) [inline include/gridtools/interface/repository/repository.hpp]:
```suggestion
```

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch icgrid

**Comment by havogt** [comment]
launch mpistrgrid

**Comment by havogt** [comment]
launch mpistrgrid

**Comment by havogt** [comment]
launch icgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch icgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch icgrid

**Pull request merged**